plugins/module_utils/network/sonic/config/ptp_port_ds/ptp_port_ds.py

```diff
@@ -0,0 +1,355 @@
+        :#rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_ptp_port_ds_facts
+        if want:
+            want = remove_empties_from_list(want)
+            normalize_interface_name(want, self._module, 'interface')
+
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            diff = get_diff(want, have, TEST_KEYS)
+            commands, requests = self._state_merged(diff)
+        elif state == 'overridden' or state == 'replaced':
+            commands, requests = self._state_replaced_or_overridden(want, have)
+
+        return commands, requests
+
+    def _state_replaced_or_overridden(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands, del_commands = [], []
+        requests, del_requests = [], []
+        state = self._module.params['state']
+        if have:
+            for have_conf in have:
+                port_num = retrieve_port_num(self._module, have_conf['interface'])
+                intf_name = have_conf['interface']
+                conf = next((cfg for cfg in want if cfg['interface'] == intf_name), None)
+                # Delete all PTP port configurations , if
+                # 1) Only the port is specified.
+                # 2) State is overridden and port is not in want
+                if (state == 'overridden' and not conf) or (conf and len(conf.keys()) == 1):
+                    del_commands.append(have_conf)
+                    del_requests.append(self.get_delete_ptp_port_request(port_num))
+                elif conf:
+                    del_command = {}
+                    for option in self.ptp_port_options:
+                        if have_conf.get(option) and option not in conf:
+                            del_command[option] = have_conf[option]
+                    if del_command:
+                        del_command['name'] = intf_name
+                        del_commands.append(del_command)
+                        del_requests.append(self.get_delete_ptp_port_request(port_num))
```

Comment:
Since only the modified config is merged, do not delete the entire interface PTP config
```suggestion
                        del_requests.extend(self.get_delete_requests(have_conf, del_command))
```